sktime/regression/tests/test_base.py

```diff
@@ -0,0 +1,421 @@
+"""Unit tests for regression base class functionality."""
+
+__author__ = ["ksharma6"]
+
+import pickle
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from sktime.regression.base import BaseRegressor
+from sktime.regression.deep_learning.base import BaseDeepRegressor
+from sktime.regression.distance_based import KNeighborsTimeSeriesRegressor
+from sktime.regression.dummy import DummyRegressor
+from sktime.utils._testing.panel import (
+    _make_panel,
+    _make_regression_y,
+    make_regression_problem,
+)
+from sktime.utils.validation._dependencies import _check_soft_dependencies
+
+
+class _DummyRegressor(BaseRegressor):
+    """Dummy regressor for testing base class fit/predict."""
+
+    def _fit(self, X, y):
+        """Fit dummy."""
+        return self
+
+    def _predict(self, X):
+        """Predict dummy."""
+        return self
+
+
+class _DummyComposite(_DummyRegressor):
+    """Dummy regressor for testing base class fit/predict."""
+
+    def __init__(self, foo):
+        self.foo = foo
+
+
+class _DummyDeepRegressorEmpty(BaseDeepRegressor):
+    """Dummy Deep Regressor for testing empty base deep class save utilities."""
+
+    def __init__(self):
+        super().__init__()
+
+    def build_model(self, input_shape, **kwargs):
+        return None
+
+    def _fit(self, X, y):
+        return self
+
+
+class _DummyDeepRegressorFull(BaseDeepRegressor):
+    """Dummy Deep Regressor to test serialization capabilities."""
+
+    def __init__(
+        self,
+        optimizer,
+    ):
+        super().__init__()
+        self.optimizer = optimizer
+
+    def build_model(self, input_shape, **kwargs):
+        return None
+
+    def _fit(self, X, y):
+        return self
+
+
+class _DummyHandlesAllInput(BaseRegressor):
+    """Dummy regressor for testing base class fit/predict."""
+
+    _tags = {
+        "capability:multivariate": True,
+        "capability:unequal_length": True,
+        "capability:missing_values": True,
+    }
+
+    def _fit(self, X, y):
+        """Fit dummy."""
+        return self
+
+    def _predict(self, X):
+        """Predict dummy."""
+        return self
+
+
+class _DummyConvertPandas(BaseRegressor):
+    """Dummy regressor for testing base class fit/predict."""
+
+    _tags = {
+        "X_inner_mtype": "nested_univ",  # which type do _fit/_predict, support for X?
+    }
+
+    def _fit(self, X, y):
+        """Fit dummy."""
+        return self
+
+    def _predict(self, X):
+        """Predict dummy."""
+        return self
+
+
+multivariate_message = r"multivariate series"
+missing_message = r"missing values"
+unequal_message = r"unequal length series"
+
+
+def test_base_regressor_fit():
+    """Test function for the BaseRegressor class fit.
+
+    Test fit. It should:
+    1. Work with 2D, 3D and DataFrame for X and nparray for y.
```

Comment:
I am not sure how exactly these are being tested in this function. Can you please explain?